sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/OtelInternalSpanDetectionUtil.java

```diff
@@ -0,0 +1,66 @@
+package io.sentry.opentelemetry;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.semconv.SemanticAttributes;
+import io.sentry.DsnUtil;
+import io.sentry.IScopes;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class OtelInternalSpanDetectionUtil {
+
+  private static final @NotNull List<SpanKind> spanKindsConsideredForSentryRequests =
+      Arrays.asList(SpanKind.CLIENT, SpanKind.INTERNAL);
+
+  @SuppressWarnings("deprecation")
+  public static boolean isSentryRequest(
+      final @NotNull IScopes scopes,
+      final @NotNull SpanKind spanKind,
+      final @NotNull Attributes attributes) {
+    if (!spanKindsConsideredForSentryRequests.contains(spanKind)) {
+      return false;
+    }
+
+    final @Nullable String httpUrl = attributes.get(SemanticAttributes.HTTP_URL);
+    if (DsnUtil.urlContainsDsnHost(scopes.getOptions(), httpUrl)) {
+      return true;
+    }
+
+    final @Nullable String fullUrl = attributes.get(SemanticAttributes.URL_FULL);
+    if (DsnUtil.urlContainsDsnHost(scopes.getOptions(), fullUrl)) {
+      return true;
+    }
+
+    // TODO [POTEL] should check if enabled but multi init with different options makes testing hard
+    // atm
+    //    if (scopes.getOptions().isEnableSpotlight()) {
+    final @Nullable String optionsSpotlightUrl = scopes.getOptions().getSpotlightConnectionUrl();
+    final @NotNull String spotlightUrl =
+        optionsSpotlightUrl != null ? optionsSpotlightUrl : "http://localhost:8969/stream";
+
+    if (containsSpotlightUrl(fullUrl, spotlightUrl)) {
+      return true;
+    }
+    if (containsSpotlightUrl(httpUrl, spotlightUrl)) {
+      return true;
+    }
+    //    }
```

Comment:
Leftover that can be deleted

Reply:
Ideally we reintroduce the `if` above, then it's needed again. We can retest after doing the alpha release.